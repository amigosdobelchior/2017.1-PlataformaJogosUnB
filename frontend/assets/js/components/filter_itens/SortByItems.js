import React from "react";
import PropTypes from "prop-types";
import {Dropdown} from "semantic-ui-react";

export default class SortByItems extends React.Component {

    constructor (props){
        super(props);
        this.state = {
            "name": "Ordernar por"
        }
    }

    listDropdownItens(){
        const rule = [{
            name: 'Nome (A-Z)',
            param: 'name',
        },{
            name: 'Nome (Z-A)',
            param: '-name',
        },{
            name: 'Mais recente',
            param: '-information__launch_year',
        },{
            name: 'Mais antigo',
            param: 'information__launch_year',
        }]
        const listDropItens = rule.map((item, i) =>
                <Dropdown.Item key={i} onClick={(e) => this.handleClick(item, e)}>
                    {item.name}
                </Dropdown.Item>
        );
        return listDropItens;
    }

    handleClick(item){
        const option = item.param;
        this.setState({ name: item.name });
        this.props.callbackParent('sortByOption', option);
    }

    render () {
        return (
            <Dropdown text={this.state.name} defaultValue='Ordernar por'>
                <Dropdown.Menu>
                    {this.listDropdownItens()}
                </Dropdown.Menu>
            </Dropdown>
        );
    }

}
SortByItems.propTypes = {
  callbackParent: PropTypes.func.isRequired
}
