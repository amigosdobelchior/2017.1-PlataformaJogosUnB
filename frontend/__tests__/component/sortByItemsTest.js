import React from 'react';
import SortByItems from '../../assets/js/components/filter_itens/SortByItems';
import {Dropdown} from "semantic-ui-react";
import renderer from 'react-test-renderer';

test('Test render SortByItems', () => {
  const component = renderer.create(
      <Dropdown text="nome" defaultValue='Ordernar por' selection>
          <Dropdown.Menu>
              "nome do item"
          </Dropdown.Menu>
      </Dropdown>
	);

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
