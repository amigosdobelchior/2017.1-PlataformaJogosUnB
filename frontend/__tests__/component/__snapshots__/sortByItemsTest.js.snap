// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test render SortByItems 1`] = `
<div
  aria-busy={undefined}
  aria-disabled={undefined}
  aria-expanded={false}
  aria-multiselectable={undefined}
  className="ui selection dropdown"
  onBlur={[Function]}
  onChange={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onMouseDown={[Function]}
  role="listbox"
  tabIndex={0}
>
  <select
    aria-hidden="true"
    multiple={undefined}
    name={undefined}
    type="hidden"
    value="Ordernar por"
  >
    <option
      value=""
    />
  </select>
  <div
    className="text"
  >
    nome
  </div>
  <i
    aria-hidden="true"
    className="dropdown icon"
  />
  <div
    className="menu transition"
  >
    "nome do item"
  </div>
</div>
`;
